storage/src/index/mod.rs

```diff
@@ -373,4 +373,27 @@ mod tests {
         assert!(iter.next().is_none());
         assert!(context.encode().contains("collisions_total 5"));
     }
+
+    #[test_traced]
+    fn test_index_remove_middle_then_next() {
```

Comment:
While working on reproducing this panic, I created this unrelated failing test (that may cause other issues).